bundles/org.openhab.binding.tibber/src/main/java/org/openhab/binding/tibber/internal/handler/TibberHandler.java

```diff
@@ -0,0 +1,482 @@
+    }
+
+    public void open() throws Exception {
+        if (isConnected()) {
+            logger.debug("Open: connection is already open");
+        }
+        logger.info("Connecting to: {}", SUBSCRIPTION_URL);
+
+        sslContextFactory.setTrustAll(true);
+        sslContextFactory.setEndpointIdentificationAlgorithm(null);
+
+        client = new WebSocketClient(sslContextFactory);
+        client.setMaxIdleTimeout(360 * 1000);
+
+        socket = new TibberWebSocketListener();
+
+        request = new ClientUpgradeRequest();
+        String token = new StringBuilder("Bearer ").append(configuration.getToken()).toString();
+        request.setHeader("Authorization", token);
+        request.setSubProtocols("graphql-subscriptions");
+
+        client.start();
+        client.connect(socket, new URI(SUBSCRIPTION_URL), request);
+    }
+
+    public void close() {
+        if (session != null) {
+            String disconnect = "{\"type\":\"connection_terminate\",\"payload\":null}";
+            try {
+                socket.sendMessage(disconnect);
+            } catch (IOException e) {
+                logger.warn("Closing websocket stream");
+            }
+            session.close();
+            session = null;
+        }
+    }
+
+    public boolean isConnected() {
+        if (session == null || !session.isOpen()) {
+            return false;
+        }
+        return connected;
+    }
+
+    @WebSocket
+    public class TibberWebSocketListener {
+
+        @OnWebSocketConnect
+        public void onConnect(Session wssession) {
+            session = wssession;
+            connected = true;
+            logger.info("Connected to Server");
+
+            String connection = "{\"type\":\"connection_init\", \"payload\":\"token=" + configuration.getToken()
+                    + "\"}";
+            try {
+                sendMessage(connection);
+            } catch (IOException e) {
+                logger.warn("Send Message Exception: ", e);
+            }
+        }
+
+        @OnWebSocketClose
+        public void onClose(int statusCode, String reason) {
+            logger.warn("Closing a WebSocket due to {}", reason);
+            session = null;
+            connected = false;
+        }
+
+        @OnWebSocketError
+        public void onWebSocketError(Throwable e) {
+            logger.error("Error during websocket communication: {}", e.getMessage(), e);
+            onClose(0, e.getMessage());
+        }
+
+        @OnWebSocketMessage
+        public void onMessage(String message) {
+            if (message.contains("connection_ack")) {
+                startSubscription();
+            } else if (message.contains("error")) {
+                close();
+                logger.warn("WebSocket Connection closed due to Connection error: {}", message);
+            } else if (!message.contains("error") && message.contains("liveMeasurement")) {
+                JsonObject Object = (JsonObject) new JsonParser().parse(message);
```

Comment:
All field and variable names should be camelcase.
```suggestion
                JsonObject object = (JsonObject) new JsonParser().parse(message);
```

Reply:
Corrected.